vllm/entrypoints/llm.py

```diff
@@ -285,7 +287,9 @@ def generate(
             considered legacy and may be deprecated in the future. You should
             instead pass them via the ``inputs`` parameter.
         """
-        if self.llm_engine.model_config.embedding_mode:
+        if self.llm_engine.model_config.model_mode in [
```

Comment:
I think whitelisting the model modes would be easier to maintain instead of blacklisting them.